codec-http2/src/main/java/io/netty/handler/codec/http2/Http2RemoteFlowController.java

```diff
@@ -53,6 +53,11 @@
 
         /**
          * Signal an error and release any retained buffers.
+         *<p>
+         * This method might be called multiple times.
```

Comment:
We might not need `complete()` at all.  IIRC, this was added to avoid the problem:

1) Flow controller calls write on the `FlowControlled`
2) `FlowControlled` closes the stream (via half-close)
3) Flow controller is notified of the closure and clears all pending writes for the stream.
4) The pendingWrite is still in the queue so its `error()` callback is invoked.

I think we could avoid this if we just always temporarily removed the pendingWrite before calling `FlowControlled.write`.  If the stream is still writable after the write, then we can restore it at the head of the queue. WDYT?

I think that simplifies the restrictions on calling `error` as well.  We still probably want to make sure that `error` is called at most once, however.